src/Collection/CollectionInterface.php

```diff
@@ -620,9 +620,10 @@ public function combine($keyPath, $valuePath, $groupPath = null);
      * whether an element is parent of another
      * @param callable|string $parentPath the column name path to use for determining
      * whether an element is child of another
+     * @param string $childrenKey The key name to use for the children
      * @return \Cake\Collection\CollectionInterface
      */
-    public function nest($idPath, $parentPath);
+    public function nest($idPath, $parentPath, $childrenKey = null);
```

Comment:
I'm pretty sure this could cause issues in any other implementations of the interface. I'm not sure there are any, but it could happen.